[BE4] 정용준

oungsi2000 opened the pull request.

oungsi2000 committed:
feat:t1~t8 통과

oungsi2000 committed:
feat:@Componet에 @Bean이 주입될 수있게

oungsi2000 committed:
fix:버그 수정

oungsi2000 committed:
fix:@Bean 싱글톤으로 보장, context name 문제 해결

jhs512 commented:
ApplicationContext.init 에서 모든 @Bean 함수와 @Compoent 클래스의 정보를 수집해 두면
그 이후에 편할 것 같습니다. 저 같은 경우 BeanDefinition 이라는 클래스를 정의해서 각 Bean 을 만드는 방법을 저장해 두었습니다.

즉 빈을 찾고 생성하는 속도를 빠르게 하려고 시도해보시면 좋을 것 같습니다,.

고생하셨습니다.

oungsi2000 committed:
fix:init 시 모든 데이터 가져오게 수정

주요 수정사항
1. init 시 모든 빈 가져옴
2. 클래스패스 탐색을 프로퍼티로 설정하여 init 시 한 번만 수행
3. genBeanByInvocation 시 다른 인스턴스를 컨텍스트에 추가하는 에러 수정

oungsi2000 commented:
> ApplicationContext.init 에서 모든 @bean 함수와 @Compoent 클래스의 정보를 수집해 두면 그 이후에 편할 것 같습니다. 저 같은 경우 BeanDefinition 이라는 클래스를 정의해서 각 Bean 을 만드는 방법을 저장해 두었습니다.
> 
> 즉 빈을 찾고 생성하는 속도를 빠르게 하려고 시도해보시면 좋을 것 같습니다,.
> 
> 고생하셨습니다.

피드백 반영하였습니다 !
